folder_paths.py

```diff
@@ -138,15 +138,20 @@ def recursive_search(directory, excluded_dir_names=None):
         excluded_dir_names = []
 
     result = []
-    dirs = {directory: os.path.getmtime(directory)}
```

Comment:
Removing this breaks the caching.

Reply:
Could you point out a test I can do for caching? I hadn't noticed anything negative on my end, but probably because I'm don't know where to look.

Would replacing that line with this allow the cache to work?

```python
    dirs = {}

    # Attempt to add the initial directory to dirs with error handling
    try:
        dirs[directory] = os.path.getmtime(directory)
    except FileNotFoundError:
        print(f"Warning: Unable to access {directory}. Skipping this path.")
```

Reply:
yes that should work.